rest/client/client.go

```diff
@@ -0,0 +1,211 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-resty/resty/v2"
+)
+
+const (
+	HeaderRequestID = "X-Request-ID"
+)
+
+// todo: add comments for godoc
+// todo: could use internal if we don't want outside packages to use this directly
+
+// Params defines an interface that path parameter and query parameter types must implement.
+type Params interface {
+	Path() map[string]string
+	Query() map[string]string
+}
+
+// HTTPBase provides functionality to make API requests via HTTP.
+type HTTPBase struct {
+	rc *resty.Client
+}
+
+type HTTPBaseConfig struct {
+	URL        string
+	Key        string
+	MaxRetries int
```

Comment:
Is this the default max retries, but the user is able to adjust this on a per call basis?

Reply:
yeah, this is just the default. for now, i'm going to hardcode some of these defaults and we can figure out which ones to add to our request options later